pkg/apis/core/helper/helpers.go

```diff
@@ -337,7 +337,16 @@ func GetShootAuditPolicyConfigMapRef(apiServerConfig *core.KubeAPIServerConfig)
 
 // HibernationIsEnabled checks if the given shoot's desired state is hibernated.
 func HibernationIsEnabled(shoot *core.Shoot) bool {
-	return shoot.Spec.Hibernation != nil && shoot.Spec.Hibernation.Enabled != nil && *shoot.Spec.Hibernation.Enabled
+	return shoot.Spec.Hibernation != nil && pointer.BoolDeref(shoot.Spec.Hibernation.Enabled, false)
+}
+
+// HibernationIsEnabled checks if the given shoot is in hibernation or is waking up.
+func IsShootInHibernation(shoot *core.Shoot) bool {
+	if shoot.Spec.Hibernation != nil && shoot.Spec.Hibernation.Enabled != nil {
+		return *shoot.Spec.Hibernation.Enabled || shoot.Status.IsHibernated
+	}
```

Comment:
Can it be done like this?
```suggestion
	if shoot.Spec.Hibernation != nil {
		return pointer.BoolDeref(shoot.Spec.Hibernation.Enabled, false) || shoot.Status.IsHibernated
	}
```

Reply:
Yes, but if `Enabled` is nil, anyways we return `status.IsHibernated` already. I don't see much benefit in changing this.